core/amber/src/main/scala/edu/uci/ics/texera/web/service/WorkflowJobService.scala

```diff
@@ -153,28 +165,30 @@ class WorkflowJobService(
     }
     resultService.attachToJob(stateStore, workflowCompiler.logicalPlan, client)
     stateStore.jobMetadataStore.updateState(jobInfo =>
-      updateWorkflowState(READY, jobInfo.withEid(workflowContext.executionID)).withErrors(Seq.empty)
+      updateWorkflowState(READY, jobInfo.withEid(workflowContext.executionID))
+        .withFatalErrors(Seq.empty)
     )
     stateStore.statsStore.updateState(stats => stats.withStartTimeStamp(System.currentTimeMillis()))
     client.sendAsyncWithCallback[Unit](
       StartWorkflow(),
-      _ => stateStore.jobMetadataStore.updateState(jobInfo => updateWorkflowState(RUNNING, jobInfo))
+      _ =>
+        stateStore.jobMetadataStore.updateState(jobInfo =>
+          if (jobInfo.state != FAILED) {
+            updateWorkflowState(RUNNING, jobInfo)
+          } else {
```

Comment:
this FAILED is not from the previous run, it is from initialization. I added this to prevent the callback from resetting the workflow state to RUNNING.